x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/use_test_query.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+/**
+ * Hook used to test the data fetching execution by returning a number of found documents
+ * Or in error in case it's failing
+ */
+export function useTestQuery(fetch: () => Promise<{ nrOfDocs: number; timeWindow: string }>) {
+  const [testQueryResult, setTestQueryResult] = useState<string | null>(null);
+  const [testQueryError, setTestQueryError] = useState<string | null>(null);
+
+  const onTestQuery = useCallback(async () => {
+    setTestQueryError(null);
+    setTestQueryResult(null);
```

Comment:
Setting state to null here leads to content jumping below test query functionality in flyout. I would suggest two solutions to prevent content jumps:
- omit setting state to null
- or set fixed height to text elements 

If we will stick to first, there is a concern about long running requests. User should see that request actually started. So, it might make sense to add some kind of loading indicator.

Reply:
Good suggestion. Always recommendable to provide feedback to the user.  I've added a text to be displayed in the loading state, also the button in now indicating it's loading.